orttraining/orttraining/python/training/onnxblock/optim/optim.py

```diff
@@ -211,3 +211,82 @@ def build(self, parameters):
         )
 
         return updated_flag_name
+
+
+class SGDOptimizer(blocks.Block):
```

Comment:
There seems to be quite an overlap between SGDOptimizer and AdamWOptimizer.
Can we factor out the common logic into a parent class `_Optimizer` or `_OptimizerBase` which gets called from both of these optimizer's build method?

Reply:
You could do something similar for the `AdamW` and `SGD` forward blocks as well.

Reply:
Creating a shared base class for the Optimizer Classes is a viable option. 
However, for the AdamW and SGD classes, their high level of customization, which is closely tied to their specific parameters and node attributes, might not be that useful and will make that class complicated.